test/distributed/org/apache/cassandra/distributed/shared/ClusterUtils.java

```diff
@@ -137,6 +137,22 @@ public static void stopUnchecked(IInstance i)
         cluster.stream().forEach(ClusterUtils::stopUnchecked);
     }
 
+    /**
+     * Create a new instance and add it to the cluster, without starting it.
+     *
+     * @param cluster to add to
+     * @param fn function to add to the config before starting
+     * @param <I> instance type
+     * @return the instance added
+     */
+    public static <I extends IInstance> I addInstance(AbstractCluster<I> cluster, Consumer<IInstanceConfig> fn)
```

Comment:
nit: I'd consider removing this and just using the overload that takes an explicit config to copy, given it's only used in `StreamCloseInMiddleTest`. With multiple DCs/racks, a method that arbitrarily adds an instance to the DC/rack of the first instance in the cluster might not be desirable.